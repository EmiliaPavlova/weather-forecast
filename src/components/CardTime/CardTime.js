import React from 'react';
import { convertToCelsius } from '../../Utils/utils';

export default function CardTime({
  time,
  icon,
  description,
  temp,
  feelsLike,
}) {

  const iconLink = `http://openweathermap.org/img/wn/${icon}@2x.png`;

  return (
    <>
      <div className=''>{time}:00</div>
      <img className='' src={iconLink} alt={description}/>
      <div className=''>{convertToCelsius(temp)}&#8451;</div>
      <div className=''>{description}</div>
      <div className=''>Feels like {convertToCelsius(feelsLike)}&#8451;</div>
    </>
  )
};