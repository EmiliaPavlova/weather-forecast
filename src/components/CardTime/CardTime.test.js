import React from 'react';
import { shallow } from 'enzyme';
import CardTime from './CardTime';

describe('CardTime component', () => {
  it('converts temp props from Kelvin to Celsius', () => {
    const wrapper = shallow(<CardTime
      time='17'
      icon='icon'
      description='sunny with clouds'
      temp={292.15}
      feelsLike={292.15}
    />);
    expect(wrapper).toMatchSnapshot();
  })
})