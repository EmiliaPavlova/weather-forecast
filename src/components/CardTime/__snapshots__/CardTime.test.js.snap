// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTime component converts temp props from Kelvin to Celsius 1`] = `
<Fragment>
  <div
    className=""
  >
    17
    :00
  </div>
  <img
    alt="sunny with clouds"
    className=""
    src="http://openweathermap.org/img/wn/icon@2x.png"
  />
  <div
    className=""
  >
    19
    ℃
  </div>
  <div
    className=""
  >
    sunny with clouds
  </div>
  <div
    className=""
  >
    Feels like 
    19
    ℃
  </div>
</Fragment>
`;
